data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/RecordDispatcherImpl.java

```diff
@@ -292,6 +294,13 @@ private ConsumerRecordContext errorTransform(final ConsumerRecordContext recordC
     extensions.put(KN_ERROR_DEST_EXT_NAME, destination);
     extensions.put(KN_ERROR_CODE_EXT_NAME, String.valueOf(response.statusCode()));
 
+    // transform all headers keys to be lowercase and filter by prefix.
+    // afterwards extract the prefix and add them to the extensions map
+    response.headers().entries().stream().collect(Collectors.toMap(entry -> entry.getKey().toLowerCase(), entry -> entry.getValue()))
```

Comment:
works, but perhaps can be written "nicer" ...

Reply:
This should be much faster and IMHO cleaner:
```java
    response.headers().forEach((k, v) -> {
      if (k.regionMatches(true, 0, EKB_ERROR_PREFIX, 0, EKB_ERROR_PREFIX.length())) { // aka startsWithIgnoreCase
        extensions.put(k.substring(EKB_ERROR_PREFIX.length()).toLowerCase(), v);
      }
    });

```